account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -141,184 +118,237 @@ def _compute_name(self):
-                            ).next_by_id(),
-                            "journal_id": provider.journal_id.id,
-                            "date": statement_date,
-                        }
-                    )
-                    statement = AccountBankStatement.with_context(
-                        journal_id=provider.journal_id.id,
-                    ).create(
-                        # NOTE: This is needed since create() alters values
-                        statement_values.copy()
-                    )
-                filtered_lines = []
-                for line_values in lines_data:
-                    date = line_values["date"]
-                    if not isinstance(date, datetime):
-                        date = fields.Datetime.from_string(date)
-
-                    if date.tzinfo is None:
-                        date = date.replace(tzinfo=utc)
-                    date = date.astimezone(utc).replace(tzinfo=None)
+                provider._process_data(data, statement_date_since, statement_date_until)
+                statement_date_since = statement_date_until
 
-                    if date < statement_date_since:
-                        if "balance_start" in statement_values:
-                            statement_values["balance_start"] = Decimal(
-                                statement_values["balance_start"]
-                            ) + Decimal(line_values["amount"])
-                        continue
-                    elif date >= statement_date_until:
-                        if "balance_end_real" in statement_values:
-                            statement_values["balance_end_real"] = Decimal(
-                                statement_values["balance_end_real"]
-                            ) - Decimal(line_values["amount"])
-                        continue
+    @api.multi
+    def _retrieve_data(self, statement_date_since, statement_date_until):
+        """Retrieve data from online provider."""
+        self.ensure_one()
+        try:
+            data = self._obtain_statement_data(
+                statement_date_since, statement_date_until
+            )
+            return data
+        except:
+            _logger.warning(
+                _("Provider %s failed to obtain statement data since %s until %s"),
+                self.name,
+                statement_date_since,
+                statement_date_until,
+                exc_info=True,
+            )
+            self.message_post(
+                body=_(
+                    "Failed to obtain statement data for period "
+                    "since %s until %s: %s. See server logs for "
+                    "more details."
+                )
+                % (
+                    statement_date_since,
+                    statement_date_until,
+                    escape(str(exc_info()[1])) or _("N/A"),
+                ),
+                subject=_("Issue with Online Bank Statement self"),
+            )
```

Comment:
This message post will be rolled back, because it is directly followed by an uncaught `raise`. Plus, after the `raise`, it will stop, and not continue with the next statement. 

If you want to do it that way, you should create a new `Environment` for the `message_post`, and then raise.

But shouldn't you do it like the original does: log and continue in case of `self.env.context.get("scheduled")`, and stop and raise to the user in case of a manual action?

Reply:
Ah no I'm wrong, you're catching it in the `_scheduled_pull` wrapper function

Reply:
Well it will be rollbacked for non scheduled sessions. The log entry will still be made. This is very usefull as the end-user receiving a message will often dismiss it without saving. For support it is still in the log then. I will move the message post so it is only done for scheduled pull's.